### ISSUE TITLE
Channeled Spells

### ISSUE THREAD

**Issue by Micha2903** [tobmaps/TrinityCore#99]
I wonder that nobody spoke about this anoing bug. For example mindflay. 

Channeling is stopped too early for me with 20% haste.

22% haste -> Mind Flay 3.0 to 2.34 seconds
-> therefore there have to be ticks every ~ 0.8 seconds

First Problem is, that mindflay (with 22% haste) ticks every ~ 0.6 second

Second Problem is that i got (also with 22% haste) 75% 3 proccs, 20% 2 proccs and 5% 1 procc.

This is for all those channeled spells like mindflay, missile barrage, arcane missile, priests hymn and so on.

I found some old tickets about it:

http://code.google.com/p/trinitycore/issues/detail?id=1052
http://code.google.com/p/trinitycore/issues/detail?id=3126

p.s. Alexander, pls, pls, pls be so kind and stop public conversations in kyrillian languages in your repo...its so hard to follow those conversations and introduce own ideas...this would be realy realy great, thanks a lot for your work

**Comment by tobmaps** [comment]
Did you tried [p118]? I've made some tests with it and seems these spells works fine in case of haste (even with high values like 3000 haste)

```
p.s. Alexander, pls, pls, pls be so kind and stop public conversations in kyrillian languages in your repo...its so hard to follow those conversations and introduce own ideas...this would be realy realy great, thanks a lot for your work
```

I'm answering to people who asked at language that they can understand :)

**Comment by Micha2903** [comment]
Ofc, I added  p118 allready. Perhabs I have to try to explain a bit better. So look:

For example we take a channeld spell X. This spell got 10 seconds for channeling time and 4 ticks. So every tick will be after 2.5 seconds. So we have ticks at 2.5, 5.0., 7.5 and finaly 10.0. At 10.0 the spell stops because his channeltime is 10 seconds. Now the problem with f.e. 20% haste

The time for the channelspell and the time between the ticks does not scale synchronus. So you have (at 20% haste f.e.) 7 seconds channel time BUT 2 seconds between every tick. So you have got ticks at 2.0, 4.0, 6.0, 8.0  but the spell stops at ~7 seconds (all just for a better understanding).

And this does not belong to p118 i guess. Its a major bug in the mechanic of calculating the time between ticks effected by haste. Dunno which of both times calculated in a right way, the channeling time or the time between ticks.

**Comment by Micha2903** [comment]
I asked some others for more information and the channeling time himself seems to scale to high with haste. Mindflay without haste has got 3 seconds channelingtime. Tested with 22% haste it used to have 2.34 seconds. But for now it breaks between 2.0 and 2.2. So the last tick/procc is missing and this effect is in all spells with the same mechanic.

p.s. why do i always klick on close and comment -.-
are you able to deactivate this feature?

**Comment by tobmaps** [comment]
Thanks for good explanation, i need to check this one...

```
are you able to deactivate this feature?
```

nope

**Comment by Wattiqelumes** [comment]
Same happens for Penance. :)

**Comment by 2010phenix** [comment]
and some players say about Vampiric tick over haste i think too
but maybe wrong remeber bug report....

**Comment by tobmaps** [comment]
fixed in official repo
